conn.go

```diff
@@ -858,6 +859,15 @@ func (c *Conn) sendFrame(ctx context.Context, fr frames.Frame, sent chan error)
 	select {
 	case c.txFrame <- frameEnvelope{Ctx: ctx, Frame: fr, Sent: sent}:
 		debug.Log(2, "TX (Conn %p): mux frame to connWriter: %s", c, fr)
+	// cancelling here should be considered safe as nothing has hit the network yet
```

Comment:
This isn't true in some cases.  Client-side close and sending of disposition frames are a few examples.  If we allow cancelling here, and it fails, then we'd need to kill the session at minimum (maybe Conn, haven't thought about this in a while).

Reply:
Some of the obvious problems:

* Disposition deletes handles and assumes it's either sent or the entire session dies.